hangupsbot/plugins/cleverbot.py

```diff
@@ -1,242 +1,26 @@
-        payload during normal operation of the bot
-        """
-        if self.asked <= 1:
-            payload = ( "stimulus={0[stimulus]}"
-                        "&cb_settings_scripting=no"
-                        "&islearning=1"
-                        "&icognoid={0[icognoid]}" ).format(self.data)
-
-            query_string = ""
-
-        else:
-            payload = ( "stimulus={0[stimulus]}"
-                        "&vText2={0[vText2]}"
-                        "&vText3={0[vText3]}"
-                        "&vText4={0[vText4]}"
-                        "&vText5={0[vText5]}"
-                        "&vText6={0[vText6]}"
-                        "&vText7={0[vText7]}"
-                        "&sessionid={0[sessionid]}"
-                        "&cb_settings_language=es"
-                        "&cb_settings_scripting=no"
-                        "&islearning={0[islearning]}"
-                        "&icognoid={0[icognoid]}" ).format(self.data)
-
-            query_string = {
-                'out' : self.lastanswer, 
-                'in' : self.data['stimulus'], 
-                'bot' : 'c', 
-                'cbsid' : self.data['sessionid'], 
-                'xai' : self.data["sessionid"][0:3], 
-                'ns' : self.asked, 
-                'al' : '', 
-                'dl' : '', 
-                'flag': '', 
-                'user' : '', 
-                'mode' : 1, 
-                't' : randint(10000, 99999)
-            }
-
-        # Generate the token
-        digest_txt = payload[9:35]
-        token = hashlib.md5(digest_txt.encode('utf-8')).hexdigest()
-        payload += "&icognocheck={}".format(token)
-
-        # Add the token to the data
-        payload = payload.encode('utf-8')
-        full_url = self.API_URL + "?" + urllib.parse.urlencode(query_string)
-        logger.debug(payload)
-        logger.debug(full_url)
-        req = urllib2.Request(full_url, payload, self.headers)
-
-        # POST the data to Cleverbot's API
-        conn = urllib2.urlopen(req)
-        resp = conn.read()
-
-        # Return Cleverbot's response
-        return resp
-
-    def _parse(self):
-        """Parses Cleverbot's response"""
-        parsed = [
-            item.split('\r') for item in self.resp.decode('utf-8').split('\r\r\r\r\r\r')[:-1]
-        ]
-
-        parsed_dict = {
-            'answer': parsed[0][0],
-            'conversation_id': parsed[0][1],
-            # 'conversation_log_id': parsed[0][2],
-        }
-        try:
-            parsed_dict['unknown'] = parsed[1][-1]
-        except IndexError:
-            parsed_dict['unknown'] = None
-        return parsed_dict
-
+cw = None
 
 def _initialise(bot):
     plugins.register_handler(_handle_incoming_message, type="message")
     plugins.register_user_command(["chat"])
     plugins.register_admin_command(["chatreset"])
-
+    api_key = bot.get_config_option("cleverbot-api-key")
```

Comment:
Abstract the checking of the api_key and creating of the CleverWrap away, so that you can run the command at the start of `_handle_incoming_message`, `chat` and `chatreset`, otherwise one will have to reload the plugin after entering an api key.